gson/src/main/java/com/google/gson/JsonElement.java

```diff
@@ -140,6 +150,23 @@ public JsonNull getAsJsonNull() {
     throw new IllegalStateException("Not a JSON Null: " + this);
   }
 
+  /**
+   * convenience method to get this element as a {@link JsonValue}. If the element is of some
+   * other type, a {@link IllegalStateException} will result. Hence it is best to use this method
+   * after ensuring that this element is of the desired type by calling {@link #isJsonValue()}
+   * first.
+   *
+   * @return get this element as a {@link JsonValue}.
+   * @throws IllegalStateException if the element is of another type.
+   * @since 2.8.7
+   */
+  public JsonValue getAsJsonValue() {
+    if (isJsonValue()) {
+      return (JsonValue) this;
+    }
+    throw new IllegalStateException("Not a JSON Null: " + this);
```

Comment:
Not sure what the "correct" exception type is here. Most of the other methods throw `UnsupportedOperationException`.